components/public-api/gitpod/experimental/v1/workspaces.proto

```diff
@@ -277,6 +289,9 @@ message WorkspaceInstanceStatus {
     // ports is the list of exposed ports in the workspace.
     repeated Port ports = 7;
 
+    // recent_folders is the opened folders inside the workspace. Used to determine the folder path to load the workspace in.
+    repeated string recent_folders = 8;
```

Comment:
Just to check, these will only ever be folders, and never files directly, right?

Reply:
Good question, there's a special case only for vscode for `.code-workspace` files which will open the folders listed inside.